Add "Manage Tags..." context menu option

AbeJellinek opened the pull request:
Fixes #2014.

Do we want a keyboard shortcut too? Or is that Zutilo territory?

AbeJellinek committed:
Add "Add Tag..." context menu option

dstillman commented on chrome/content/zotero/zoteroPane.js:
```diff
@@ -3505,6 +3509,23 @@ var ZoteroPane = new function()
+	this.addTagToSelectedItems = function () {
+		let items = Zotero.Items.keepParents(this.getSelectedItems());
+		let out = { value: null };
+		let result = Services.prompt.prompt(
```
People are going to want to use this in lieu of dragging, particularly if there's a keyboard shortcut, so I think this should be a nice big popup panel that does tag autocomplete:

https://github.com/zotero/zotero/blob/0785b3a0864fc2dfce6ab53a88c08f6d9d1cd5a7/chrome/content/zotero/containers/tagsBoxContainer.jsx#L71-L99

For the input, we have this — with a `getSuggestions` prop — in React, which is what we use in the tagsBox component:

https://github.com/zotero/zotero/blob/master/chrome/content/zotero/components/form/input.jsx

The prompt works, of course, but this hasn't existed for 15 years, so I think we should do it right — major new version, major new way of assigning quickly tags to one or more items with a keyboard shortcut.

We might also consider doing this for collections, with results showing the "path" to subcollections, instead of the hierarchical menu, which gets pretty unwieldy in a huge library. Or at least offering it as a keyboard-based alternative.

(There's no rush on this.)

AbeJellinek commented on chrome/content/zotero/zoteroPane.js:
```diff
@@ -3505,6 +3509,23 @@ var ZoteroPane = new function()
+		);
+		if (result) {
+			items.forEach(item => item.addTag(out.value));
+		}
```
Thanks! I always forget to save after adding tags.

AbeJellinek committed:
Replace simple prompt with a tags box popup

This reworks tagsBox.js a bit, but behavior should be exactly the same
when only viewing one item (which is always the case in the item box).
The external API remains the same if you're just setting/retrieving a
single item.

AbeJellinek commented:
OK, reworked into something quite a bit fancier using the tags box CE.

(Tags box supporting multiple items might be helpful if we ever decide to implement batch item editing.)

AbeJellinek commented:
Looks like this:

<img width="416" alt="image" src="https://user-images.githubusercontent.com/1770299/180494967-036a62a9-71ea-4c98-be9b-86ac52087fec.png">

1. Is this along the lines of what we'd want?
2. Should it update if you select a new item/set of items, Quick Look-style?
3. Should it have a close button or is pressing Escape obvious enough?

Ray-Eldath commented:
this seems solve https://forums.zotero.org/discussion/69095/batch-removal-of-tags I suppose? I can take on this if the original author is busy with other stuff. hope the review could finished and get this released soon. as one comment said:

![image](https://user-images.githubusercontent.com/7938763/198963284-5458d563-f1d4-40e8-9907-9573f04580a5.png)

dstillman commented:
@Ray-Eldath: Please don't post to these issues, which are for developers. This issue has nothing to do with that thread.
